File: app/src/main/java/cm/aptoide/pt/comment/data/CommentsResponseModel.java
package cm.aptoide.pt.comment.data;

import java.util.Collections;
import java.util.List;

public class CommentsResponseModel {
  private final List<Comment> comments;
  private final int offset;
  private final boolean loading;

  public CommentsResponseModel(List<Comment> comments, int offset) {
    this.comments = comments;
    this.offset = offset;
    this.loading = false;
  }

  public CommentsResponseModel(boolean loading) {
    this.loading = loading;
    this.comments = Collections.emptyList();
    this.offset = -1;
  }

  public List<Comment> getComments() {
    return comments;
  }

  public int getOffset() {
    return offset;
  }

  public boolean isLoading() {
    return loading;
  }
}

File: app/src/main/java/cm/aptoide/pt/comments/refactor/data/CommentsResponseModel.kt
package cm.aptoide.pt.comments.refactor.data

import cm.aptoide.pt.editorial.epoxy.comments.CommentFilters
import java.util.*

data class CommentsResponseModel(val comments: List<Comment>, val offset: Int,
                                 val total: Int, val filters: CommentFilters,
                                 val loading: Boolean) {
  constructor(loading: Boolean, filters: CommentFilters) : this(Collections.emptyList(), -1, -1,
      filters,
      loading)

  constructor(comments: List<Comment>, offset: Int, total: Int, filters: CommentFilters) : this(
      comments,
      offset, total, filters,
      false)

  fun hasMore(): Boolean {
    return total > offset
  }
}